examples/fabric/dcgan/train_fabric.py

```diff
@@ -0,0 +1,268 @@
+"""
+DCGAN - Accelerated with Lightning Fabric
+
+Code adapted from the official PyTorch DCGAN tutorial:
+https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html
+"""
+import os
+import time
+from pathlib import Path
+
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.optim as optim
+import torch.utils.data
+import torchvision.transforms as transforms
+import torchvision.utils
+from torchvision.datasets import CelebA
+
+from lightning.fabric import Fabric, seed_everything
+
+# Root directory for dataset
+dataroot = "data/"
+# Number of workers for dataloader
+workers = os.cpu_count()
+# Batch size during training
+batch_size = 128
+# Spatial size of training images
+image_size = 64
+# Number of channels in the training images
+nc = 3
+# Size of z latent vector (i.e. size of generator input)
+nz = 100
+# Size of feature maps in generator
+ngf = 64
+# Size of feature maps in discriminator
+ndf = 64
+# Number of training epochs
+num_epochs = 5
+# Learning rate for optimizers
+lr = 0.0002
+# Beta1 hyperparameter for Adam optimizers
+beta1 = 0.5
+# Number of GPUs to use
+num_gpus = 1
+
+
+def main():
```

Comment:
just thinking to make all the default/constants above as arguments here so that with a simple use of [`Fire`](https://github.com/google/python-fire) for CLI creation all these parameters can be exposed in CLI :rabbit:

Reply:
This is the standard dcgan example from pytorch, I do not plan to refactor it in such ways.